src/compiler/diagnosticMessages.json

```diff
@@ -1847,6 +1847,10 @@
         "category": "Error",
         "code": 2676
     },
+    "A type guard's type must be assignable to its parameter's type.": {

```

Comment:
I'm thinking we could change this to `"A type predicate's type must be assignable to its parameter's type."`. Any strong feelings @RyanCavanaugh @sandersn?